TACC/Core-CMS
# feat: grid editor group and permissions

## Overview & Changes

Add a "Grid Editor (Basic)" and "Grid Editor (Advanced)".

## Related

- [CMD-189](https://tacc-main.atlassian.net/browse/CMD-189)

## Testing

1. Create/Open a Core-CMS instance.
2. In the docker, run:
    - `python manage.py set_group_perms grid_editor_basic`
    - `python manage.py set_group_perms grid_editor_advanced`
3. In CMS admin UI, verify two groups are created:
    - "Grid Editor (Basic)"
    - "Grid Editor (Advanced)"
4. Verify the groups have these permissions:
    - "Grid Editor (Basic)" can view & change containers & rows & columns.
    - "Grid Editor (Advanced)" can view & change containers & rows & columns.

## UI

| before | after |
| - | - |
| <img width="920" alt="before" src="https://github.com/user-attachments/assets/4f1f16fc-0833-444d-a07d-5b7cf4e4b802"> | <img width="920" alt="after" src="https://github.com/user-attachments/assets/a8ff834b-ccd2-447b-8855-89c112f4baab"> |

| basic | advanced |
| - | - |
| <img width="920" alt="after -basic" src="https://github.com/user-attachments/assets/72226b9f-5b59-438d-8a28-cc5d9ab0b280"> | <img width="920" alt="after - advanced" src="https://github.com/user-attachments/assets/bfaf1e65-0093-4762-81af-b48f8fdd3afe"> |
